array-reduce/reduce.js

```diff
@@ -0,0 +1,70 @@
+const numbersArray = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const accountArray = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traitsArray = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+function reduce(array, reducer, initialValue) {
```

Comment:
The `reducer` should be a callback function, not a string. In side of the `reduce` function you should loop through the array, and call the `reducer` callback function inside the loop for each value in the array. You want to call it and handle the returned value in a way that the end result mimics what the actual `reduce` method does.